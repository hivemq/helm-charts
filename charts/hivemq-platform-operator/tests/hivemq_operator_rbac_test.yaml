suite: test HiveMQ Platform Operator RBAC
templates:
  - rbac.yml
tests:

  - it: with disabled RBAC creation
    set:
      rbac.create: false
    asserts:
      - notFailedTemplate: { "rbac.yml" }
      - hasDocuments:
          count: 0

  - it: with default platform service account settings
    asserts:
      - exists:
          path: rules
      - contains:
          path: rules
          content:
            apiGroups: [ "" ]
            resources: [ "serviceaccounts" ]
            verbs: [ "get", "list", "watch", "create", "patch", "update", "delete" ]
      - contains:
          path: rules
          content:
            apiGroups: [ "rbac.authorization.k8s.io" ]
            resources: [ "roles", "rolebindings" ]
            verbs: [ "get", "list", "watch", "create", "patch", "update", "delete" ]
      - contains:
          path: rules
          content:
            apiGroups: [ "rbac.authorization.k8s.io" ]
            resources: [ "clusterroles", "clusterrolebindings" ]
            verbs: [ "get", "list", "watch" ]

  - it: with platform service account create and validate enabled
    set:
      hivemqPlatformServiceAccount.create: true
      hivemqPlatformServiceAccount.validate: true
    asserts:
      - contains:
          path: rules
          content:
            apiGroups: [ "" ]
            resources: [ "serviceaccounts" ]
            verbs: [ "get", "list", "watch", "create", "patch", "update", "delete" ]

  - it: with platform service account create disabled
    set:
      hivemqPlatformServiceAccount.create: false
      hivemqPlatformServiceAccount.validate: true
    asserts:
      - contains:
          path: rules
          content:
            apiGroups: [ "" ]
            resources: [ "serviceaccounts" ]
            verbs: [ "get", "list", "watch" ]

  - it: with platform service account validate disabled
    set:
      hivemqPlatformServiceAccount.create: true
      hivemqPlatformServiceAccount.validate: false
    asserts:
      - contains:
          path: rules
          content:
            apiGroups: [ "" ]
            resources: [ "serviceaccounts" ]
            verbs: [ "get", "list", "watch", "create", "patch", "update", "delete" ]

  - it: with platform service account create and validate disabled
    set:
      hivemqPlatformServiceAccount.create: false
      hivemqPlatformServiceAccount.validate: false
    asserts:
      - notContains:
          path: rules
          content:
            apiGroups: [ "" ]
            resources: [ "serviceaccounts" ]
            verbs: [ "get", "list", "watch" ]
      - notContains:
          path: rules
          content:
            apiGroups: [ "" ]
            resources: [ "serviceaccounts" ]
            verbs: [ "get", "list", "watch", "create", "patch", "update", "delete" ]

  - it: with platform service account permissions create and validate enabled
    set:
      hivemqPlatformServiceAccount.permissions.create: true
      hivemqPlatformServiceAccount.permissions.validate: true
    asserts:
        - contains:
            path: rules
            content:
              apiGroups: [ "rbac.authorization.k8s.io" ]
              resources: [ "roles", "rolebindings" ]
              verbs: [ "get", "list", "watch", "create", "patch", "update", "delete" ]
        - contains:
            path: rules
            content:
              apiGroups: [ "rbac.authorization.k8s.io" ]
              resources: [ "clusterroles", "clusterrolebindings" ]
              verbs: [ "get", "list", "watch" ]

  - it: with platform service account permissions create disabled
    set:
      hivemqPlatformServiceAccount.permissions.create: false
      hivemqPlatformServiceAccount.permissions.validate: true
    asserts:
      - contains:
          path: rules
          content:
            apiGroups: [ "rbac.authorization.k8s.io" ]
            resources: [ "roles", "rolebindings" ]
            verbs: [ "get", "list", "watch" ]
      - contains:
          path: rules
          content:
            apiGroups: [ "rbac.authorization.k8s.io" ]
            resources: [ "clusterroles", "clusterrolebindings" ]
            verbs: [ "get", "list", "watch" ]

  - it: with platform service account permissions validate disabled
    set:
      hivemqPlatformServiceAccount.permissions.create: true
      hivemqPlatformServiceAccount.permissions.validate: false
    asserts:
      - contains:
          path: rules
          content:
            apiGroups: [ "rbac.authorization.k8s.io" ]
            resources: [ "roles", "rolebindings" ]
            verbs: [ "get", "list", "watch", "create", "patch", "update", "delete" ]
      - notContains:
          path: rules
          content:
            apiGroups: [ "rbac.authorization.k8s.io" ]
            resources: [ "clusterroles", "clusterrolebindings" ]
            verbs: [ "get", "list", "watch" ]

  - it: with platform service account permissions create and validate disabled
    set:
      hivemqPlatformServiceAccount.permissions.create: false
      hivemqPlatformServiceAccount.permissions.validate: false
    asserts:
      - notContains:
          path: rules
          content:
            apiGroups: [ "rbac.authorization.k8s.io" ]
            resources: [ "roles", "rolebindings" ]
            verbs: [ "get", "list", "watch", "create", "patch", "update", "delete" ]
      - notContains:
          path: rules
          content:
            apiGroups: [ "rbac.authorization.k8s.io" ]
            resources: [ "roles", "rolebindings" ]
            verbs: [ "get", "list", "watch" ]
      - notContains:
          path: rules
          content:
            apiGroups: [ "rbac.authorization.k8s.io" ]
            resources: [ "clusterroles", "clusterrolebindings" ]
            verbs: [ "get", "list", "watch" ]
