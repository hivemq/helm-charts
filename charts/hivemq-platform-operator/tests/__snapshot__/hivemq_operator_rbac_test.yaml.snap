with default values:
  1: |
    - apiGroups:
        - hivemq.com
      resources:
        - hivemq-platforms
        - hivemq-platforms/status
        - hivemq-platforms/finalizers
      verbs:
        - get
        - list
        - watch
        - patch
        - update
        - create
        - delete
    - apiGroups:
        - apiextensions.k8s.io
      resources:
        - customresourcedefinitions
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
    - apiGroups:
        - rbac.authorization.k8s.io
      resources:
        - clusterroles
        - clusterrolebindings
      verbs:
        - get
        - list
        - watch
    - apiGroups:
        - ""
      resources:
        - configmaps
        - events
        - pods
        - secrets
        - services
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
        - delete
    - apiGroups:
        - apps
      resources:
        - statefulsets
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
        - delete
    - apiGroups:
        - ""
      resources:
        - pods/exec
      verbs:
        - create
        - watch
        - get
    - apiGroups:
        - ""
      resources:
        - serviceaccounts
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
        - delete
    - apiGroups:
        - rbac.authorization.k8s.io
      resources:
        - roles
        - rolebindings
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
        - delete
with namespaces:
  1: |
    - apiGroups:
        - hivemq.com
      resources:
        - hivemq-platforms
        - hivemq-platforms/status
        - hivemq-platforms/finalizers
      verbs:
        - get
        - list
        - watch
        - patch
        - update
        - create
        - delete
    - apiGroups:
        - apiextensions.k8s.io
      resources:
        - customresourcedefinitions
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
    - apiGroups:
        - rbac.authorization.k8s.io
      resources:
        - clusterroles
        - clusterrolebindings
      verbs:
        - get
        - list
        - watch
  2: |
    - apiGroups:
        - ""
      resources:
        - configmaps
        - events
        - pods
        - secrets
        - services
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
        - delete
    - apiGroups:
        - apps
      resources:
        - statefulsets
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
        - delete
    - apiGroups:
        - ""
      resources:
        - pods/exec
      verbs:
        - create
        - watch
        - get
    - apiGroups:
        - ""
      resources:
        - serviceaccounts
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
        - delete
    - apiGroups:
        - rbac.authorization.k8s.io
      resources:
        - roles
        - rolebindings
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
        - delete
  3: |
    - apiGroups:
        - ""
      resources:
        - configmaps
        - events
        - pods
        - secrets
        - services
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
        - delete
    - apiGroups:
        - apps
      resources:
        - statefulsets
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
        - delete
    - apiGroups:
        - ""
      resources:
        - pods/exec
      verbs:
        - create
        - watch
        - get
    - apiGroups:
        - ""
      resources:
        - serviceaccounts
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
        - delete
    - apiGroups:
        - rbac.authorization.k8s.io
      resources:
        - roles
        - rolebindings
      verbs:
        - get
        - list
        - watch
        - create
        - patch
        - update
        - delete
